### ISSUE TITLE
DauxIO throwing 500 error after upgrading nginx server to php 7

### ISSUE THREAD

**Issue by tylerfoulkes** [justinwalsh/daux.io#385]
I upgraded my nginx server on Digital Ocean to the new ubuntu release which runs PHP 7 and I am now getting an http 500 error when I navigate to my documentation. I checked the nginx logs and I am getting the following error:

```
2016/07/10 06:36:08 [error] 11785#11785: *7581 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught
 Error: Call to undefined function League\CommonMark\mb_strcut() in phar:///var/www/docs/daux.phar/vendor/leagu
e/commonmark/src/Cursor.php:317
```

Any help would be greatly appreciated!

**Comment by JayLancaster** [comment]
I could be wrong, but i think if you do an install php7-mbstring via apt-get should fix the issue.

**Comment by tylerfoulkes** [comment]
@JayLancaster That worked thanks! However the documentation links don't point to anything, on reload the new address appears in the address bar but the page just reloads the original landing page, any idea what could be causing this? I no longer get server errors.

**Comment by JayLancaster** [comment]
Once again, a touch shaky on this, but here is my best attempt. 

nano /etc/nginx/sites-available/default

scroll down and you should see the settings for root and index. 

for the most part it will be set to /usr/share/nginx/html

How is  your documentation available? is it static pages or a full dump?

**Comment by tylerfoulkes** [comment]
I believe it's a full dump meaning the whole DauxIO directory structure is located on the server.

**Comment by JayLancaster** [comment]
in your docs folder, in config.json

Can you find this section here and check that it mirriors mine?
"live": {
        "inherit_index": true,
        "clean_urls": true

**Comment by tylerfoulkes** [comment]
Yes my config.json reads exactly that.

**Comment by tylerfoulkes** [comment]
I have the same directory structure working fine on my local environment which uses Apache so I am thinking it must be an Nginx config issue as that is what the live server is using. Could it be a rewrite issue?

**Comment by tylerfoulkes** [comment]
It is definitely a server issue but I am not quite sure what could be wrong, I have a Wordpress site running on the same server with the same server block configs and the links work fine. Any ideas as to why this might be?

**Comment by tylerfoulkes** [comment]
Ok so I kind of have it fixed but its still not completely working. I finally figured out that it was a rewrite rule issue and something about going from php 5 to php 7 made support for my old rewrite rule non-existent. I found an older support issue on here about nginx rewrite rules and used it: [(https://github.com/justinwalsh/daux.io/issues/99)]. It works for urls such as mysite.com/Overview however as soon as I navigate to mysite.com/Overview/Subdoc I lose all styling and only content is shown and then when I try click the Overview link to go back I get mysite.com/Overview/Subdoc/Overview. I feel like I am really close, the rewrite rule just needs to be slightly adjusted.

Below is my current rewrite rule in my server block:

'''
location / {
        if (!-e $request_filename){
            rewrite ^(.*)$ /index.php$1;
        }
    }
'''

Any help would be awesome!
